Tests/powershell-yaml.Tests.ps1

```diff
@@ -94,6 +94,23 @@ InModuleScope $moduleName {
         }
 
         Context "Test array handling under various circumstances." {
+
+            It "Should ConvertTo empty array." {
+                $str = ConvertTo-Yaml @()
+                $str.Trim() | Should -Be "[]"
+            }
+
+            It "Should ConvertTo empty array with -OutFile." {
+                $tempFile = [System.IO.Path]::GetTempFileName()
+                ConvertTo-Yaml @() -OutFile $tempFile -Force
+                (Get-Content $tempFile).Trim() | Should -Be "[]"
```

Comment:
I think you can do something like:

```powershell
$data = @()
$tempFile = [System.IO.Path]::GetTempFileName()
 ConvertTo-Yaml $data -OutFile $tempFile -Force
$readData = ConvertFrom-Yaml (Get-Content -Raw $tempFile)
($data -eq $readData) | Should -Be $true
```

roughly. This way you test both writing to file and that reading the yaml you just saved, can be read back. Note, comparing 2 arrays using ```-eq``` might not work.

Reply:
Okay, I will add the `ConvertFrom` part after reading the file.

Reply:
In Pester there is `TestDrive:\` file drive setup for every `Describe`/`Context` you can use that. It setups in a random folder so there is no clash between multiple blocks and it will properly clean up. :) 

The `($data -eq $readData) | Should -Be $true` would not work, the right side will enumerate when you give it an array and pass only values that equal to the right side. So `1,2,3 -eq 1,2,3` returns `$null`, because no single item on the left side equals to the whole array on the right side. 

For the same reason comparing two empty arrays returns `$null` so `(@() -eq @()) | Should -Be $true` will never pass. 

Comparing empty arrays is a bit difficult because it's hard to see what is going on, and you need to think about nulls and so on. I would either use multiple assertions to check that both sides are not $null, and have your target type, and Length/Count 1.